File: LeetCode/C++/200-299/217.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> repeat;
        for(int i = 0; i < nums.size();++i){
            if(repeat.count(nums[i])){
                return true;
            }
            
            repeat.insert(nums[i]);
        }
        
        return false;
    }
};

int main() {
	vector<int> nums{1,2,3,1};
	Solution sol;
	cout << sol.containsDuplicate(nums) << endl;
	return 0;
}

File: LeetCode/C#/200-299/217.cs
using System;
using System.Collections.Generic;

namespace LeetCode217
{
    public class Solution {
		public bool ContainsDuplicate(int[] nums) {
			HashSet<int> repeat = new HashSet<int>();
			for(int i = 0; i < nums.Length;++i){
				if(repeat.Contains(nums[i])){
					return true;
				}
				
				repeat.Add(nums[i]);
			}
			
			return false;
		}
	}

    class Program
    {
        static void Main(string[] args)
        {
            int[] nums = new int[]{1,2,3,1};
            Solution sol = new Solution();
            Console.WriteLine(sol.ContainsDuplicate(nums));
            Console.Read();
        }
    }
}
